paimon-core/src/main/java/org/apache/paimon/table/system/ManifestsTable.java

```diff
@@ -211,7 +213,11 @@ private static List<ManifestFileMeta> allManifests(FileStoreTable dataTable) {
             }
             snapshot = snapshotManager.snapshot(snapshotId);
         } else if (snapshotId == null) {
```

Comment:
here, Condition 'snapshotId == null' is always 'true'

Reply:
yes ，change to else